### ISSUE TITLE
[bug] "Ignore keyboard layout" may be reset when upgraded to v1.99.0

### ISSUE THREAD

**Issue by Mahno9** [gdh1995/vimium-c#703]
**Vimium C stops working accidentally**

After update to Version 1.99.0. I've note that the extension may stop working accidentally for no apparent reason.

**To Reproduce**

1. Work with your browser with Vimium C 1.99.0 extension installed;
2. (Probably) Open a link in a new tab with "F" and back to source page with Ctrl+Shift+Tab;
3. Note that in some moment navigation keys and links following shortcuts stop working on a page.

To make the extension work again it may help to restart browser, re-enable extension or export-import settings for it .

**Browser, OS and Vimium C versions**

- Browser name: Yandex browser
- Browser version: 22.7.3.822 (64-bit) (from "chrome://version": Yandex | 102.0.5005.822 (64-разрядная версия))
- Vimium C version: 1.99.0
- OS name and version: Windows 10 Version 21H2 (Build 19044.1889)

**Comment by gdh1995** [comment]
Do you mean sometimes it doesn't work on all tabs? Or just a few "long lived" tabs?

When this happens, is there any shortcut (key mapping) of Vimium C still work in a newly created tab?

Will some keys of Vimium C work again if you click the "source page" where you think Vimium C has been broken?

**Comment by Mahno9** [comment]
Yes, sometimes it doesn't work on all tabs. Regardless of lifetime.

I can't use any of my usual key mappings. I mean navigation (j, k, d, u), link following (f, F), tab closing (x). It feels like Vimium C is disabled on pages.

I'm not sure I understand what do you mean under "click the "source page". Is it about switch back to page where Vimium C has worked but stops then? The extension stops working on all pages I have openned in my browser. Also, as I remember, on all new pages I'm openning (I'll double check this in next time the problem reproduced).

I would happy to share some logs from Vimium C if there are way to take it.

**Comment by gdh1995** [comment]
Thanks for your feedback.

I forgot to ask a keypoint: what's your keyboard layout and Input Method (IME) ? Will Vimium C work again if you switch to a English QWERTY layout?

You said it might recover if you "export-import settings", then I wonder:
1. now that you can export settings of Vimium C, does it mean you can open Vimium C Options?
- 1. Can you open the Options page with neither restarting your browser nor re-enabling the whole extension?
2. when you are on Vimium C Options page and haven't re-importing settinsg, does any key mapping work on the tab?
3. What do you see on Vimium C Options? Does your configuration show well, or are all options showing their default values?
4. after re-importing settings, will Vimium C work on the Options tab at once?
5. Will Vimium C work on other tabs at once without reloading tabs?

Vimium C has a background process, so its log can be seen if:
1. open `chrome://extensions`, and enable the `Developer mode`
2. find the Vimium C card, click it, and then click a link named `background page` in the `Inspect views` section
3. on a opened DevTools window, click the `Console` panel name
4. then all log messages are listed on the panel

**Comment by Mahno9** [comment]
Usually I'm using Russian keyboard layout. I've tried to reproduce the problem with this keyboard layout and that's it. Switching back to QWERTY fixes the problem. 

I was surprised because previously I had no problem with the non-QWERTY layout. I've noted that in Vimium C options I have the option "Always ignore keyboard layout" disabled now, but I remember how I found that this option fixes similar issue for me for several months ago. 
Probably, update have disabled this option and layout issue came back to me. Seems like I was just switching the layout during switching tabs, and while I've exported/imported options.

Thanks a lot for your help! And thank you for this incredible product!

**Comment by gdh1995** [comment]
Um yes I updated the logic, some options were removed and a new complex option was added. while I forgot to address this change on the upgrading notification.

I once wrote some code to auto compute the new option's value from those old options, but it seems there're bugs in it. Sorry for the inconvenience.

Then what's your current settings of `Keyboard layout rules` ? Will it works on your Russian keyboard layout, if you enable like this:
![image](https://user-images.githubusercontent.com/5547703/184833093-0f928275-ab8a-4639-a841-6218a94e85f8.png)
